functionaltests/internal/kbclient/client_test.go

```diff
@@ -0,0 +1,150 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kbclient_test
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/dnaeon/go-vcr/cassette"
+	"github.com/dnaeon/go-vcr/recorder"
+	"github.com/itchyny/gojq"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/apm-server/functionaltests/internal/kbclient"
+)
+
+// getHttpClient instantiate a http.Client backed by a recorder.Recorder to be used in testing
+// scenarios.
+// To update any fixture, delete it from the filesystem and run the test again.
+// NOTE: always remember to call (recorder.Recorder).Stop() in your test case.
+func getHttpClient(t *testing.T) (*recorder.Recorder, *http.Client) {
+	t.Helper()
+
+	var err error
+	var rec *recorder.Recorder
+	var hc *http.Client
+
+	tr := http.DefaultTransport
+	rec, err = recorder.NewAsMode(path.Join("testdata", "fixtures", t.Name()), recorder.ModeReplaying, tr)
```

Comment:
Possible to make t.Cleanup call rec.Stop, instead of calling it in every test? I'm not familiar with how the vcr works.

Reply:
🤦 yes obviously! Thanks for catching this.

Reply:
addressed in [e85c8df](https://github.com/elastic/apm-server/pull/15818/commits/e85c8dfc4f366ccfb65e3ac1561d37be89a72707)